tests/handlers/test_koji.py

```diff
@@ -48,6 +48,143 @@ def test_build_complete(mock_koji_cs, mock_getBuild_two):
+
+    commit = DistGitCommit.nodes.get_or_none(hash_='c75628d32fc0c8b1bd5e1e7b047538a25c983569')
+
+    build.commit.is_connected(commit)
+
+
+@mock.patch('koji.ClientSession')
+def test_build_canceled(mock_koji_cs, mock_getBuild_canceled, kb_canceled):
+    """Test the Koji handler when it recieves a new build failed message."""
+    mock_koji_session = mock.Mock()
+    mock_koji_session.getBuild.return_value = mock_getBuild_canceled
+    mock_koji_cs.return_value = mock_koji_session
+
+    orig_build = kb_canceled
+    assert orig_build.state == 4
+
+    with open(path.join(message_dir, 'koji', 'build_canceled.json'), 'r') as f:
+        msg = json.load(f)
+    assert KojiHandler.can_handle(msg) is True
+    handler = KojiHandler(config)
+    handler.handle(msg)
+
+    build = KojiBuild.nodes.get_or_none(id_='751869')
+    assert build is not None
+    assert build.completion_time == datetime(2018, 8, 18, 15, 4, 47, tzinfo=pytz.utc)
+    assert build.creation_time == datetime(2018, 8, 15, 13, 47, 43, tzinfo=pytz.utc)
+    assert build.epoch is None
+    assert build.extra == ('{"source": {"original_url": "git://pkgs.domain.com/rpms/gcc#c72278'
+                           '1ead329f9b76751ba89ee7451109cebde6"}}')
+    assert build.id_ == '751869'
+    assert build.name == 'devtoolset-8-gcc'
+    assert build.release == '2.2.el7'
+    assert build.version == '8.2.1'
+    assert build.start_time == datetime(2018, 8, 15, 13, 47, 43, tzinfo=pytz.utc)
+    assert build.state == 4
+
+    commit = DistGitCommit.nodes.get_or_none(hash_='c722781ead329f9b76751ba89ee7451109cebde6')
+
+    build.commit.is_connected(commit)
+
+
+@mock.patch('koji.ClientSession')
+def test_build_deleted(mock_koji_cs, mock_getBuild_deleted, kb_deleted):
+    """Test the Koji handler when it recieves a new build failed message."""
+    mock_koji_session = mock.Mock()
+    mock_koji_session.getBuild.return_value = mock_getBuild_deleted
+    mock_koji_cs.return_value = mock_koji_session
+
+    orig_build = kb_deleted
+    assert orig_build.state == 1
+
+    with open(path.join(message_dir, 'koji', 'build_deleted.json'), 'r') as f:
+        msg = json.load(f)
+    assert KojiHandler.can_handle(msg) is True
+    handler = KojiHandler(config)
+    handler.handle(msg)
+
+    build = KojiBuild.nodes.get_or_none(id_='623989')
+    assert build is not None
+    assert build.completion_time == datetime(2017, 12, 7, 22, 32, 47, tzinfo=pytz.utc)
+    assert build.creation_time == datetime(2017, 12, 7, 22, 21, 54, tzinfo=pytz.utc)
+    assert build.epoch is None
+    assert build.extra == ('{"source": {"original_url": "git://pkgs.domain.com/rpms/gcc#d50611'
+                           '6380ccc12760fd5a42b1c52201f7ffe510"}}')
+    assert build.id_ == '623989'
+    assert build.name == 'gnome-terminal'
+    assert build.release == '1.el8'
+    assert build.version == '3.26.1'
+    assert build.start_time == datetime(2017, 12, 7, 22, 21, 54, tzinfo=pytz.utc)
+    assert build.state == 2
```

Comment:
Optional: Since you are really just checking the state transition occurred, you can remove all these other asserts above.